containers/proxy-tftpd-image/uyuni-configure.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+import re
+import yaml
+import sys
+
+config_path = "/etc/uyuni/"
+
+# read from file
+with open(config_path + "config.yaml") as source:
+    config = yaml.safe_load(source)
+   
+    server_version = config.get("server_version")
+    # Only check version for SUSE Manager, not Uyuni
+    matcher = re.fullmatch(r"([0-9]+\.[0-9]+\.)[0-9]+", server_version)
+    if matcher:
+        major_version = matcher.group(1)
+        container_version = subprocess.run(["rpm", "-q", "--queryformat", "%{version}", "spacewalk-proxy-common"],
+                stdout=subprocess.PIPE, universal_newlines=True).stdout
+        if not container_version.startswith(major_version):
+            print("FATAL: Proxy container image version (%s) doesn't match server major version (%s)".format(
+                container_version, major_version), file=sys.stderr)
+            sys.exit(1)
+    
+    tftp_config = "/etc/sysconfig/tftp"
+    tftp_root = "/srv/tftpboot"
+    with open(tftp_config, "w") as file:
+        file.write(f'''# Automatically generated Uyuni Proxy Server configuration file.
+        TFTP_USER="tftp"
+        TFTP_OPTIONS="{config.get('tftp_options', '')} "
+        TFTP_DIRECTORY="{tftp_root}"''')
+
+    os.system("chown root:tftp {}".format(tftp_config))
+    os.system("chmod 640 {}".format(tftp_config))
```

Comment:
IIUC `subprocess.run` is meant to be used over the `os.system` only if you are about to handle the return value of the command you're firing, am I missing something? If we are not handling the return, we can simply use `os.system` from my POV

Reply:
I wasn't aware of this. Do you have the documentation where this is stated. I was can read at the top of the Python Docs that the `subprocess` module intends to replace the `os.system` module: https://docs.python.org/3.8/library/subprocess.html

Reply:
I only remember that from a verbal knowledge I heard about. If the official documentation says it to prefer one over another, that is probably more relevant then. The only other point I could see on the discussion table is the backward compatibility keep using `os.system`, but I don't think it is an issue here. `os.system` seems to use the `shell` by default, in `subprocess` it has to be declared, other than that I don't have any preference. Just in case we opt for `subprocess`, I'd then suggest to refactor all the other `uyuni-configure.py` in the other images as well. Is so, @aaannz maybe it can be done in a separate dedicated commit for all outside of this scope.

Reply:
I will resolve this, as per @ncounter 's correct comment this is out of scope for this PR